vsock-proxy/enclave/src/dsm_key_config.rs

```diff
@@ -25,34 +25,38 @@ use sdkms::SdkmsClient;
 use url;
 
 use crate::certificate::CertificateResult;
-use crate::file_system::find_env_or_err;
 
 const GCM_TAG_LEN_BITS: usize = 128;
 const SOBJECT_LIST_LIMIT: usize = 10;
-pub const DEFAULT_DSM_ENDPOINT: &str = "https://amer.smartkey.io/";
 const OVERLAY_FS_SECURITY_OBJECT_PREFIX: &str = "fortanix-overlayfs-security-object-build-";
-pub const DEFAULT_DSM_APP_ENDPOINT: &str = "https://apps.amer.smartkey.io/";
 const TLS_MIN_VERSION: Version = Version::Tls1_2;
 
 struct ClientWithKey {
     overlayfs_key: Sobject,
     dsm_client: SdkmsClient,
 }
 
-fn dsm_create_client(env_vars: &[(String, String)], auth_cert: Option<&mut CertificateResult>) -> Result<SdkmsClient, String> {
-    info!("Looking for env variables needed to create DSM client.");
-    let api_key = find_env_or_err("FS_API_KEY", env_vars).unwrap_or("".to_string());
+/// Information needed to connect to DSM as a client
+pub struct ClientConnectionInfo<'a> {
+    pub fs_api_key: Option<String>,
+    pub auth_cert: Option<&'a mut CertificateResult>,
+    pub dsm_url: String,
```

Comment:
Do these fields need to be `pub`? Having them as `pub` means they can be changed anywhere, at any point, even outside of this crate. That's not "wrong", but makes it more difficult to reason about them. Having only private fields, and a public constructor, makes it obvious that a value of this struct is created once, and then won't be changed.

Reply:
I can update it to pub(crate), they need to be accessible in the vsock-proxy/enclave crate only.